### DISCUSSION TITLE
Success install bot on android TV box 😁

### DISCUSSION THREAD

**Discussion by khanfar** [n3d1117/chatgpt-telegram-bot#288] [Show and tell]
![20230430_190406.jpg](https://user-images.githubusercontent.com/16803586/235363547-f9168738-3d2d-437a-99f7-84c37eba1214.jpg)

![20230430_190422.jpg](https://user-images.githubusercontent.com/16803586/235363548-64510d78-9e4c-47fe-b884-f5d75683131a.jpg)

![20230430_190413.jpg](https://user-images.githubusercontent.com/16803586/235363549-7c2f3855-ba2f-4251-aea5-50e6bbbe830f.jpg)

**Comment by khanfar** [comment]
I'm successfully install this bit in Amlogic android TV box S905W , I'm happy to share everything how I do that ...

**Comment by khanfar** [comment]
Now I put this small TV box down table running 7/24 on 5v source 🌟

**Comment by khanfar** [comment]
What I'm doing is install custom rom very easy called slimbox from slimboxtv.ru then install custom kernel Linux based up it called Armbian

**Comment by khanfar** [comment]
The TV box called amlogic X96mini and have S905W chip with 2gb ram ddr3

**Comment by khanfar** [comment]
![20230430_191056.jpg](https://user-images.githubusercontent.com/16803586/235363838-cafb4457-9bfb-469a-83d1-3a50342b7a0f.jpg)

This is vnc viewer to control from pc , or you can control from ssh like putty or from your phone vnc viewer or hock hdmi cable directly to your tv

**Comment by khanfar** [comment]
![20230430_191656.jpg](https://user-images.githubusercontent.com/16803586/235364073-9b5ff501-56b8-4ce2-8220-06a51ac2c472.jpg)

![Uploading 20230430_191501.jpg …]()
Here the bot is running too

**Comment by khanfar** [comment]
![20230430_191800.jpg](https://user-images.githubusercontent.com/16803586/235364098-4e3ac1ee-e567-44a4-81c9-a07677210d72.jpg)

In the left side , this TV box have chatgpt bot running 7 24 , very small and can acsess it from anywhere too

**Comment by khanfar** [comment]
I make the bot running automatically when tv box boot

**Comment by khanfar** [comment]
Any one have this TV box I'm happy to share kernel and custom rom to do that

**Comment by khanfar** [comment]
this is armbian ubuntu os link to download : https://drive.google.com/file/d/18EObf3QuYFdNl-1gDr2mSQfpRi1OuMhX/view

**Comment by khanfar** [comment]
and you need this for fire armbian in 8gb memory card to put in tv box and boot it , https://www.balena.io/etcher

**Comment by khanfar** [comment]
for booting you need small slim thing to put inside av port to push small hidden button for booting

**Comment by khanfar** [comment]
i do first by install slim box .img in stock firmware , you can download that from slimbox website  https://slimboxtv.ru/

**Comment by khanfar** [comment]
then i setup armbian linux up it to make it sucsess https://www.armbian.com/

**Comment by khanfar** [comment]
for slim box :   after download the image you want ,, its .img .  you have two way to fire on tv box
1- direct from pc you need usb to usb cable and Amlogic_USB_Burning_Tool_v2.2.4 .
2- using sd card 8gb formatted to fat32 and software to burn to it called burn card maker

**Comment by khanfar** [comment]
for making bot run on boot i do this :      
Create the autostart directory if it doesn't exist:

mkdir -p ~/.config/autostart

Create a new desktop entry file for your bot in XFCE:

nano ~/.config/autostart/chatgpt_bot_xfce.desktop

now add this : 

[Desktop Entry]
Type=Application
Exec=xfce4-terminal -e "bash -c 'python3 /home/mwk/chatgpt-telegram-bot/bot/main.py; read -n1 -s'"
Name=ChatGPT Telegram Bot
Comment=Start ChatGPT Telegram Bot in a terminal

then save and reboot

**Comment by khanfar** [comment]
for env file you can just from file manager do view then show hidden files . now you can see it and edit as you like

**Comment by khanfar** [comment]
this is amlogic usb burning tool :   https://androiddatahost.com/khfj4    to fire slimbox.image directly in tv  box using usb to usb cable you can make easy

**Comment by khanfar** [comment]
this is burn card maker if you want to burn slim box rom .image directly to usb storage micro 8gb fat32 , you can use this tool : https://androidmtk.com/amlogic-burn-card-maker

**Comment by khanfar** [comment]
remember after install and run slim box rom in tv box . then you install armbian linux up that to work

**Comment by khanfar** [comment]
the boot method push hidden button inside av port then put 5v power source and still push for 10 to 15 seconds then release

**Comment by khanfar** [comment]
![20230430_194827.jpg](https://user-images.githubusercontent.com/16803586/235365576-349e3d38-d46a-4f96-9a69-66396de51e87.jpg)

**Comment by khanfar** [comment]
this is slim box .img i use :  https://drive.google.com/file/d/1iEVN1x4Ubgz9XO4rJvn2wbDRm63W5GCM/view?usp=share_link

**Comment by khanfar** [comment]
slim box image name :    sbx_x96mini_pie_v12_1.img

**Comment by khanfar** [comment]
first fire slim box image , then after boot the tv box do armbian fire , then done

**Comment by khanfar** [comment]
when first run armbian you can go to application from menu up left then settings thin click on armbian config then you can setup ssh , remote acsess , settings every thin you want , then you can open terminal and do update and install everything you want

**Comment by khanfar** [comment]
then i do this to install bot :  

sudo apt-get update
sudo apt-get upgrade
sudo apt-get install git python3 python3-pip
git clone https://github.com/n3d1117/chatgpt-telegram-bot
cd chatgpt-telegram-bot
pip3 install -r requirements.txt
then put your env file in that directory or copy your actual bot working files over that directory

**Comment by khanfar** [comment]
for ffmpeg you can do this :   

sudo apt-get update
sudo apt-get install ffmpeg
nano ~/.bashrc
export PATH=$PATH:/path/to/ffmpeg
source ~/.bashrc
nano ~/.bashrc
export PATH=$PATH:/home/mwk/chatgpt-telegram-bot/bot
source ~/.bashrc
sudo mv /home/mwk/chatgpt-telegram-bot/bot/ffmpeg /usr/local/bin/

**Comment by khanfar** [comment]
When boot armbian from micro USB and then you want to install it in actual TV box internal storage you do this : 

sudo su 
cd /root
ls
sudo ./install-aml.sh

**Comment by n3d1117** [comment]
Awesome @khanfar thanks for sharing!

**Reply by khanfar** [reply]
Thanks . I'm just only issue error see in bot terminal,  but bot works , please see this error , if can help how to fix these errors.. see this ![20230430_134527.jpg](https://user-images.githubusercontent.com/16803586/235493583-8d07c752-5f60-49ac-a474-77ed59d942b4.jpg)

**Reply by khanfar** [reply]
This become when I copy all .py files from pc bot to android TV box and I change the token not sucsess to remove these errors from appearing,  and I'm not run the bot in pc anymore.  This become when I make bot run automatically on boot and I'm make sure just one bot is run on boot and not run it in windows pc anymore.  Please help me with that issue ..

**Reply by khanfar** [reply]
I'm just change the bit token in env file , I keep financial records folder as it is , so I copy everything in bot folder from my pc to tv box . And still get that error on terminal.
